man/coredns-metadata.7

```diff
@@ -0,0 +1,44 @@
+.\" generated with Ronn/v0.7.3
+.\" http://github.com/rtomayko/ronn/tree/0.7.3
+.
+.TH "COREDNS\-METADATA" "7" "July 2018" "CoreDNS" "CoreDNS plugins"
+.
+.SH "NAME"
+\fImetadata\fR \- enable a meta data collector\.
+.
+.SH "DESCRIPTION"
+By enabling \fImetadata\fR any plugin that implements metadata\.Provider interface \fIhttps://godoc\.org/github\.com/coredns/coredns/plugin/metadata#Provider\fR will be called for each DNS query, at beginning of the process for that query, in order to add it\'s own meta data to context\.
+.
+.P
+The meta data collected will be available for all plugins, via the Context parameter provided in the ServeDNS function\. The package (code) documentation has examples on how to inspect and retrieve metadata a plugin might be interested in\.
+.
+.P
+The meta data is added by setting a label with a value in the context\. These labels should be named \fBplugin/NAME\fR, where \fBNAME\fR is something descriptive\. The only hard requirement the \fImetadata\fR plugin enforces is that the labels contains a slash\. See the documentation for \fBmetadata\.SetValueFunc\fR\.
+.
+.P
+The value stored is a string\. The empty string signals "no meta data"\. See the documentation for \fBmetadata\.ValueFunc\fR on how to retrieve this\.
```

Comment:
this info: `empty string signals "no meta data"`
we have 3 cases:
- Provider advertise a Label and when retrieving the value, that value is not empty (expected regular case)
- Provider advertise a Label and when retrieving the value, that value is empty
- Provider do not advertise a Label 

Which one do we call "no meta data", view from the consumer ?

Consumer knows if Label is advertised with ValueFunc not NIL
if consumer can retrieve a value for a Label with ValueFunc, then there is a meta data .. and maybe that value is empty.

IMO,  the "no meta data" case is when the Label is not advertised (ValueFunc return a nil)